### PR TITLE
Support Orleans 4.0.0-preview2

### LINKED ISSUE #369: Dependency on Microsoft.Orleans.Runtime.Abstraction breaks on Orleans 4.0 preview 2
The `OrleansDashboard.Core` nuget package has a dependency on `Microsoft.Orleans.Runtime.Abstractions ≥ 3.6.0` which is causing me some issues when installing OrleansDashboard in a project that depends on Orleans 4.0 preview 2:

```
SiloBuilder.cs(8, 36): [CS0433] The type 'ISiloBuilder' exists in both 'Orleans.Runtime.Abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null' and 'Orleans.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null'
```

### PR DESCRIPTION
**Pull request by tanordheim**
This upgrades OrleansDashboard to support the latest Orleans 4.0.0-preview2 release. Quick summary of the changes:

- Some Orleans persistence and reminder components are split out into separate packages with the latest release, added those to the relevant projects.
- Microsoft.Orleans.Runtime.Abstractions is replaced with Microsoft.Orleans.OrleansRuntime.

I have also removed the `DashboardTelemetryConsumer` as it implements the `IMetricTelemetryConsumer` which is no longer available in Orleans after the project migrating to using System.Diagnostics for instrumentation in https://github.com/dotnet/orleans/pull/7854.

Fixes #369

### CODE REVIEW

**Comment by SebastianStehle** [comment]
Thanks a lot.

The question is:
Shall we remove the whole metrics system, because it is useless now or do we implement it with an OTLP adapter?

**Comment by tanordheim** [comment]
@SebastianStehle question is what the metrics system is used for I guess. From how I read the code, the `DashboardTelemetryConsumer` was collecting some of the metric data exposed by Orleans but nothing was really displaying it anywhere - hence removing it. I guess if its still needed we could attempt to rebuild that feature using [MeterListener](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meterlistener?view=net-6.0) perhaps.

**Comment by richorama** [comment]
The `DashboardTelemetryConsumer` data is shown on the Silo page on the "Silo Counters" panel.

**Comment by SebastianStehle** [comment]
So it could be implemented with a otlp adapter. They even support batching, which should make the code easier.

**Comment by tanordheim** [comment]
I've added a new `DashboardTelemetryExporter`, replacing the old `DashboardTelemetryConsumer`, that exports batches of metrics to the `ISiloGrain` the same way the old consumer did. It runs as a batched job every second via OpenTelemetry.

I wasn't quite sure where we wanted to initialize this, what I ended up doing was to put it into the `Dashboard` startup task. Adding it to the ServiceCollection could also be an option but given that the [OpenTelemetry.Extensions.Hosting](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting/) package is still in prerelease it didn't feel right to take that dependency on.

File: OrleansDashboard/Implementation/DashboardTelemetryExporter.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                Current = current;
+            }
+
+            public Value<T> Update(T newValue)
+            {
+                return new Value<T>(Current, newValue);
+            }
+        }
+
+        private readonly ConcurrentDictionary<string, Value<double>> metrics = new ConcurrentDictionary<string, Value<double>>();
```
Comment (SebastianStehle) [inline OrleansDashboard/Implementation/DashboardTelemetryExporter.cs:39]: Is this needed? It should be a normal dictionary in a local variable now.

**Review by SebastianStehle — CHANGES_REQUESTED** [review]

File: OrleansDashboard/Implementation/DashboardTelemetryExporter.cs (hunk 1 above)
Comment (tanordheim) [reply OrleansDashboard/Implementation/DashboardTelemetryExporter.cs]: True, I was contemplating the same but ended up just reusing the code from the old consumer. Corrected it now :)

**Comment by SebastianStehle** [comment]
LGTM

**Comment by richorama** [comment]
Thank you!

**Comment by tanordheim** [comment]
Thanks! Any chance a new preview release can be built from these packages for those running the latest Orleans-preview? Would be excellent to have the dashboard easily available there as well :)

**Pull request merged**